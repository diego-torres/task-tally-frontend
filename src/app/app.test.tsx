import * as React from 'react';
import App from '@app/index';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

describe('App tests', () => {
  test('should render landing page when unauthenticated', () => {
    const { asFragment } = render(<App />);

    expect(screen.getByRole('heading', { name: 'Task Tally' })).toBeVisible();
    expect(screen.getByRole('link', { name: /sign in with google/i })).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a nav-toggle button when authenticated', () => {
    render(<App mockAuthenticated />);

    expect(screen.getByRole('button', { name: 'Global navigation' })).toBeVisible();
  });

  it('should expand the sidebar on larger viewports when authenticated', () => {
    render(<App mockAuthenticated />);

    window.dispatchEvent(new Event('resize'));

    expect(screen.getByRole('link', { name: 'Home' })).toBeVisible();
  });

  it('should include Git SSH keys in the navigation when authenticated', () => {
    render(<App mockAuthenticated />);

    window.dispatchEvent(new Event('resize'));

    expect(screen.getByRole('link', { name: 'Git SSH keys' })).toBeVisible();
  });

  it('should hide the sidebar when clicking the nav-toggle button', async () => {
    const user = userEvent.setup();

    render(<App mockAuthenticated />);

    window.dispatchEvent(new Event('resize'));
    const button = screen.getByRole('button', { name: 'Global navigation' });
    expect(screen.getByRole('link', { name: 'Home' })).toBeVisible();

    await user.click(button);

    expect(screen.queryByRole('link', { name: 'Home' })).not.toBeInTheDocument();
  });
});
