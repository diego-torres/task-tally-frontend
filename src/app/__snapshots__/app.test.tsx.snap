// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should render default App component 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-page"
  >
    <div
      class="pf-v6-c-skip-to-content"
    >
      <a
        aria-disabled="false"
        class="pf-v6-c-button pf-m-primary"
        data-ouia-component-id="OUIA-Generated-Button-primary-1"
        data-ouia-component-type="PF6/Button"
        data-ouia-safe="true"
        href="#primary-app-container"
      >
        <span
          class="pf-v6-c-button__text"
        >
          Skip to Content
        </span>
      </a>
    </div>
    <header
      class="pf-v6-c-masthead pf-m-display-inline-on-md"
    >
      <div
        class="pf-v6-c-masthead__main"
      >
        <span
          class="pf-v6-c-masthead__toggle"
        >
          <button
            aria-disabled="false"
            aria-label="Global navigation"
            class="pf-v6-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF6/Button"
            data-ouia-safe="true"
            type="button"
          >
            <span
              class="pf-v6-c-button__icon"
            >
              <svg
                aria-hidden="true"
                class="pf-v6-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                />
              </svg>
            </span>
          </button>
        </span>
        <div
          class="pf-v6-c-masthead__brand"
          data-codemods="true"
        >
          <div
            style="display: flex; align-items: center;"
          >
            <img
              alt="Task Tally Logo"
              src="test-file-stub"
              style="height: 40px; margin-right: 12px;"
            />
            <span
              style="font-size: 1.8rem; font-weight: 700;"
            >
              Task Tally
            </span>
          </div>
        </div>
      </div>
    </header>
    <div
      aria-hidden="false"
      class="pf-v6-c-page__sidebar pf-m-expanded"
      id="page-sidebar"
    >
      <div
        class="pf-v6-c-page__sidebar-body"
      >
        <nav
          aria-label="Global"
          class="pf-v6-c-nav"
          data-ouia-component-id="OUIA-Generated-Nav-1"
          data-ouia-component-type="PF6/Nav"
          data-ouia-safe="true"
          id="nav-primary-simple"
        >
          <ul
            class="pf-v6-c-nav__list"
            id="nav-list-simple"
            role="list"
          >
            <li
              class="pf-v6-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-1"
              data-ouia-component-type="PF6/NavItem"
              data-ouia-safe="true"
            >
              <a
                aria-current="page"
                class="pf-v6-c-nav__link pf-m-current active"
                data-discover="true"
                href="/"
                style="display: flex; align-items: center; gap: 8px;"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="margin-right: 4px;"
                  viewBox="0 0 576 512"
                  width="1em"
                >
                  <path
                    d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                  />
                </svg>
                Home
              </a>
            </li>
            <li
              class="pf-v6-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-2"
              data-ouia-component-type="PF6/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-v6-c-nav__link"
                data-discover="true"
                href="/templates"
                style="display: flex; align-items: center; gap: 8px;"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="margin-right: 4px;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                  />
                </svg>
                Templates
              </a>
            </li>
            <li
              class="pf-v6-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-3"
              data-ouia-component-type="PF6/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-v6-c-nav__link"
                data-discover="true"
                href="/proposals"
                style="display: flex; align-items: center; gap: 8px;"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="margin-right: 4px;"
                  viewBox="0 0 576 512"
                  width="1em"
                >
                  <path
                    d="M572.694 292.093L500.27 416.248A63.997 63.997 0 0 1 444.989 448H45.025c-18.523 0-30.064-20.093-20.731-36.093l72.424-124.155A64 64 0 0 1 152 256h399.964c18.523 0 30.064 20.093 20.73 36.093zM152 224h328v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v278.046l69.077-118.418C86.214 242.25 117.989 224 152 224z"
                  />
                </svg>
                Proposals
              </a>
            </li>
            <li
              class="pf-v6-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-4"
              data-ouia-component-type="PF6/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-v6-c-nav__link"
                data-discover="true"
                href="/git-ssh-keys"
                style="display: flex; align-items: center; gap: 8px;"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v6-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="margin-right: 4px;"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M802,320 C748.980664,320 706,277.019336 706,224 C706,170.980664 748.980664,128 802,128 C855.019336,128 898,170.980664 898,224 C898,277.019336 855.019336,320 802,320 M704,0 C527.3,0 384,143.3 384,320 C383.937788,357.490503 390.505571,394.696657 403.4,429.9 L0,824.1 L0,1024 L192,1024 L192,896 L320,896 L320,768 L448,768 L597,622 C596.906403,621.881923 596.838304,621.745723 596.8,621.6 C631.220126,633.811107 667.47802,640.034477 704,640 C880.7,640 1024,496.7 1024,320 C1024,143.3 880.7,0 704,0"
                  />
                </svg>
                Git SSH keys
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <div
      class="pf-v6-c-page__main-container"
    >
      <main
        class="pf-v6-c-page__main"
        id="primary-app-container"
        tabindex="-1"
      >
        <section
          class="pf-v6-c-page__main-section"
        >
          <div
            class="pf-v6-c-page__main-body"
          >
            <div
              class="pf-v6-l-flex pf-m-align-items-center"
            >
              <div
                class=""
              >
                <img
                  alt="Task Tally logo"
                  src="test-file-stub"
                  style="max-width: 150px;"
                />
              </div>
              <div
                class=""
              >
                <h1
                  class="pf-v6-c-title pf-m-4xl"
                  data-ouia-component-id="OUIA-Generated-Title-1"
                  data-ouia-component-type="PF6/Title"
                  data-ouia-safe="true"
                >
                  Task Tally
                </h1>
                <p>
                  An estimation tool based on "The art of project estimation" book. This tool allows users to create consulting estimates.
                </p>
              </div>
            </div>
          </div>
        </section>
        <section
          class="pf-v6-c-page__main-section pf-m-secondary"
        >
          <div
            class="pf-v6-c-page__main-body"
          >
            <div
              class="pf-v6-l-flex pf-m-column pf-m-align-items-center"
            >
              <h2
                class="pf-v6-c-title pf-m-lg"
                data-ouia-component-id="OUIA-Generated-Title-2"
                data-ouia-component-type="PF6/Title"
                data-ouia-safe="true"
                style="text-align: center; margin-bottom: 1rem;"
              >
                Templates & Proposals Overview
              </h2>
              <svg
                height="260"
                style="max-width: 100%; height: auto; margin-bottom: 1rem;"
                viewBox="0 0 800 260"
                width="800"
              >
                <rect
                  fill="#e3f2fd"
                  height="80"
                  rx="15"
                  stroke="#1976d2"
                  stroke-width="2"
                  width="240"
                  x="3"
                  y="40"
                />
                <text
                  fill="#1976d2"
                  font-size="18"
                  font-weight="bold"
                  text-anchor="middle"
                  x="140"
                  y="70"
                >
                  Templates
                </text>
                <text
                  fill="#1976d2"
                  font-size="13"
                  text-anchor="middle"
                  x="122"
                  y="95"
                >
                  Reusable groups of tasks, outcomes, etc.
                </text>
                <polygon
                  fill="#1976d2"
                  points="250,80 290,80 290,70 320,90 290,110 290,100 250,100"
                />
                <text
                  fill="#1976d2"
                  font-size="12"
                  text-anchor="middle"
                  x="285"
                  y="65"
                >
                  can create
                </text>
                <rect
                  fill="#fff3e0"
                  height="80"
                  rx="15"
                  stroke="#f57c00"
                  stroke-width="2"
                  width="240"
                  x="320"
                  y="40"
                />
                <text
                  fill="#f57c00"
                  font-size="18"
                  font-weight="bold"
                  text-anchor="middle"
                  x="440"
                  y="70"
                >
                  Proposals
                </text>
                <text
                  fill="#f57c00"
                  font-size="13"
                  text-anchor="middle"
                  x="440"
                  y="95"
                >
                  Estimate project effort
                </text>
                <text
                  fill="#f57c00"
                  font-size="13"
                  text-anchor="middle"
                  x="440"
                  y="112"
                >
                  Composed of templates + custom tasks
                </text>
                <rect
                  fill="#fce4ec"
                  height="60"
                  rx="12"
                  stroke="#c2185b"
                  stroke-width="2"
                  width="210"
                  x="350"
                  y="140"
                />
                <text
                  fill="#c2185b"
                  font-size="15"
                  font-weight="bold"
                  text-anchor="middle"
                  x="460"
                  y="170"
                >
                  Additional Tasks/Elements
                </text>
                <text
                  fill="#c2185b"
                  font-size="12"
                  text-anchor="middle"
                  x="460"
                  y="190"
                >
                  Specific to each project proposal
                </text>
                <polygon
                  fill="#c2185b"
                  points="460,140 460,130 455,130 460,120 465,130 460,130"
                />
              </svg>
              <div
                style="max-width: 500px; text-align: center;"
              >
                <p
                  style="font-size: 1rem;"
                >
                  <strong>
                    Templates
                  </strong>
                   describe reusable groups of tasks, outcomes, and more.
                  <br />
                  <strong>
                    Proposals
                  </strong>
                   estimate project effort, combining one or more templates and additional tasks unique to the project.
                </p>
              </div>
            </div>
          </div>
        </section>
        <section
          class="pf-v6-c-page__main-section"
        >
          <div
            class="pf-v6-c-page__main-body"
          >
            <div
              class="pf-v6-l-flex pf-m-justify-content-center"
            >
              <div
                class=""
              >
                <h2
                  class="pf-v6-c-title pf-m-lg"
                  data-ouia-component-id="OUIA-Generated-Title-3"
                  data-ouia-component-type="PF6/Title"
                  data-ouia-safe="true"
                  style="text-align: center; margin-bottom: 1rem;"
                >
                  How Task Tally Works
                </h2>
                <div
                  class="pf-v6-l-flex pf-m-space-items-lg"
                >
                  <div
                    class=""
                  >
                    <div
                      style="text-align: center;"
                    >
                      <svg
                        fill="none"
                        height="64"
                        viewBox="0 0 64 64"
                        width="64"
                      >
                        <circle
                          cx="32"
                          cy="32"
                          fill="#E0E7FF"
                          r="32"
                        />
                        <text
                          fill="#3C3C3C"
                          font-size="24"
                          text-anchor="middle"
                          x="32"
                          y="38"
                        >
                          1
                        </text>
                      </svg>
                      <h3
                        class="pf-v6-c-title pf-m-md"
                        data-ouia-component-id="OUIA-Generated-Title-4"
                        data-ouia-component-type="PF6/Title"
                        data-ouia-safe="true"
                      >
                        Define Tasks
                      </h3>
                      <p
                        style="max-width: 120px; margin: 0px auto;"
                      >
                        List all project tasks and phases.
                      </p>
                    </div>
                  </div>
                  <div
                    class=""
                  >
                    <div
                      style="text-align: center;"
                    >
                      <svg
                        fill="none"
                        height="64"
                        viewBox="0 0 64 64"
                        width="64"
                      >
                        <circle
                          cx="32"
                          cy="32"
                          fill="#C7F9E9"
                          r="32"
                        />
                        <text
                          fill="#3C3C3C"
                          font-size="24"
                          text-anchor="middle"
                          x="32"
                          y="38"
                        >
                          2
                        </text>
                      </svg>
                      <h3
                        class="pf-v6-c-title pf-m-md"
                        data-ouia-component-id="OUIA-Generated-Title-5"
                        data-ouia-component-type="PF6/Title"
                        data-ouia-safe="true"
                      >
                        Estimate Effort
                      </h3>
                      <p
                        style="max-width: 120px; margin: 0px auto;"
                      >
                        Use 3-point estimates to assess LOE and risk.
                      </p>
                    </div>
                  </div>
                  <div
                    class=""
                  >
                    <div
                      style="text-align: center;"
                    >
                      <svg
                        fill="none"
                        height="64"
                        viewBox="0 0 64 64"
                        width="64"
                      >
                        <circle
                          cx="32"
                          cy="32"
                          fill="#FFD6E0"
                          r="32"
                        />
                        <text
                          fill="#3C3C3C"
                          font-size="24"
                          text-anchor="middle"
                          x="32"
                          y="38"
                        >
                          3
                        </text>
                      </svg>
                      <h3
                        class="pf-v6-c-title pf-m-md"
                        data-ouia-component-id="OUIA-Generated-Title-6"
                        data-ouia-component-type="PF6/Title"
                        data-ouia-safe="true"
                      >
                        Model Teams
                      </h3>
                      <p
                        style="max-width: 120px; margin: 0px auto;"
                      >
                        Assign consultants, teams, and skill levels.
                      </p>
                    </div>
                  </div>
                  <div
                    class=""
                  >
                    <div
                      style="text-align: center;"
                    >
                      <svg
                        fill="none"
                        height="64"
                        viewBox="0 0 64 64"
                        width="64"
                      >
                        <circle
                          cx="32"
                          cy="32"
                          fill="#FFFACD"
                          r="32"
                        />
                        <text
                          fill="#3C3C3C"
                          font-size="24"
                          text-anchor="middle"
                          x="32"
                          y="38"
                        >
                          4
                        </text>
                      </svg>
                      <h3
                        class="pf-v6-c-title pf-m-md"
                        data-ouia-component-id="OUIA-Generated-Title-7"
                        data-ouia-component-type="PF6/Title"
                        data-ouia-safe="true"
                      >
                        Analyze & Adjust
                      </h3>
                      <p
                        style="max-width: 120px; margin: 0px auto;"
                      >
                        Run scenarios, adjust scope, and optimize cost.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
        <section
          class="pf-v6-c-page__main-section"
        >
          <div
            class="pf-v6-c-page__main-body"
          >
            <div>
              <h2
                class="pf-v6-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-8"
                data-ouia-component-type="PF6/Title"
                data-ouia-safe="true"
              >
                Ever wanted to propose a Container Adoption Journey, but just a little different?
              </h2>
              <p>
                We can do that, in a fraction of the time.
              </p>
              <h2
                class="pf-v6-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-9"
                data-ouia-component-type="PF6/Title"
                data-ouia-safe="true"
              >
                What is a 3 point estimate?
              </h2>
              <ul
                class="pf-v6-c-list"
              >
                <li
                  class=""
                >
                  <span>
                    An analytical and quantitative approach to estimate the Level of Effort (LOE) for projects
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    Based on industry best practices
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    Reduces risk that projects will be under or over sized
                  </span>
                </li>
              </ul>
              <h2
                class="pf-v6-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-10"
                data-ouia-component-type="PF6/Title"
                data-ouia-safe="true"
              >
                Architects model:
              </h2>
              <ul
                class="pf-v6-c-list"
              >
                <li
                  class=""
                >
                  <span>
                    The amount of time each task will take
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    Allocate tasks to different phases of the project
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    And/or different teams working in parallel
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    How many consultants, with what titles and skill levels, are on each team
                  </span>
                </li>
              </ul>
              <h2
                class="pf-v6-c-title pf-m-xl"
                data-ouia-component-id="OUIA-Generated-Title-11"
                data-ouia-component-type="PF6/Title"
                data-ouia-safe="true"
              >
                Allows us to quickly:
              </h2>
              <ul
                class="pf-v6-c-list"
              >
                <li
                  class=""
                >
                  <span>
                    Start with a standard solution
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    Add multiple modules and/or solutions together
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    Modify, remove, or add new tasks not in the standard solutions
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    Run “what-if” scenarios
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    How much would it cost if we had a larger/smaller team?
                  </span>
                </li>
                <li
                  class=""
                >
                  <span>
                    Which tasks/phases are the expensive ones, and how could we change scope to reduce cost?
                  </span>
                </li>
              </ul>
            </div>
          </div>
        </section>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
